packages/babel-parser/src/tokenizer/index.js

```diff
@@ -651,25 +688,12 @@ export default class Tokenizer extends LocationParser {
 
   getTokenFromCode(code: number): void {
     switch (code) {
-      case charCodes.numberSign:
-        if (this.state.pos === 0 && this.readToken_interpreter()) {
-          return;
-        }
+      // The interpretation of a number sign "#" depends on whether it is
+      // followed by an identifier or not.
```

Comment:
I think that we can remove this comment, since the `readToken_numberSign` always return `tt.hash` now.